features/api-layer/dao-common/src/main/java/org/opennms/features/apilayer/dao/AlarmDaoImpl.java

```diff
@@ -83,4 +92,101 @@ public Optional<Alarm> getAlarmWithHighestSeverity(NodeRef nodeRef) {
             return Optional.of(alarm);
         });
     }
+
+    @Override
+    public Optional<Alarm> getAlarmForTicket(final String ticketId) {
+        final Criteria criteria = new CriteriaBuilder(OnmsAlarm.class)
+                .and(Restrictions.eq("tticketId", ticketId))
+                .limit(1)
+                .toCriteria();
+        return sessionUtils.withReadOnlyTransaction(() -> {
+            final List<OnmsAlarm> matching = alarmDao.findMatching(criteria);
+            if (matching.isEmpty()) {
+                return Optional.empty();
+            }
+            final Alarm alarm = ModelMappers.toAlarm(matching.get(0));
+            return Optional.of(alarm);
+        });
+    }
+
+    @Override
+    public void setTicketState(final State state, final int... alarmIds) {
+        sessionUtils.withTransaction(() -> {
+            Arrays.stream(alarmIds).boxed().map(alarmDao::get).forEach(alarm -> {
+                alarm.setTTicketState(ModelMappers.fromTicketState(state));
+                alarmDao.saveOrUpdate(alarm);
```

Comment:
Should also call `AlarmEntityNotifier#didChangeTicketStateForAlarm` after `saveOrUpdate`.